### ISSUE TITLE
[cpu inductor] where is silently incorrect when SIMD code is generated.

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#100800]
### 🐛 Describe the bug

```python
import torch

input_tensor = torch.ones(3, 3)

def f(x):
    return torch.where(torch.ones_like(x).to(torch.bool), torch.zeros_like(x), torch.ones_like(x)* 2)

res1 = f(input_tensor)
print(res1)

jit_func = torch.compile(f)
res2 = jit_func(input_tensor)
print(res2)

```

Output
```
tensor([[0., 0., 0.],
        [0., 0., 0.],
        [0., 0., 0.]])
tensor([[2., 2., 2.],
        [2., 2., 2.],
        [2., 2., 0.]])
```

Reason:
Implementation of where relies on `blendv` where MSB of the mask element should be 0 for first element of the packed vector to be copied.
https://github.com/pytorch/pytorch/blob/8d56b0a5b57cf3e82402556ceb5c7080c0f9d5b6/torch/_inductor/codegen/cpp.py#L572-L573

blendv: https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-8/mm256-blendv-ps.html

Found in https://github.com/pytorch/pytorch/pull/100799#issuecomment-1537136218

### Versions

master

cc @soumith @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire
